pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go

```diff
@@ -55,6 +55,86 @@ type KeycloakAPIRealm struct {
 	// TODO: change to values and use kubebuilder default annotation once supported
 	// +optional
 	AdminEventsDetailsEnabled *bool `json:"adminEventsDetailsEnabled,omitempty"`
+
+	// Authentication flows
+	// +optional
+	AuthenticationFlows *[]KeycloakAPIAuthenticationFlow `json:"authenticationFlows,omitempty"`
+
+	// Authenticator config
+	// +optional
+	AuthenticatorConfig []KeycloakAPIAuthenticatorConfig `json:"authenticatorConfig"`
+}
+
+type KeycloakAPIAuthenticationFlow struct {
+	// Alias
+	Alias string `json:"alias"`
+
+	// Authentication executions TODO: really optional?
+	// +optional
+	AuthenticationExecutions *[]KeycloakAPIAuthenticationExecution `json:"authenticationExecutions"`
+
+	// Built in
+	// +optional
+	BuiltIn *bool `json:"builtIn,omitempty"`
+
+	// Description
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	// ID
+	// +optional
+	ID string `json:"id,omitempty"`
+
+	// Provider ID
+	// +optional
+	ProviderID string `json:"providerId,omitempty"`
+
+	// Top level
+	// +optional
+	TopLevel bool `json:"topLevel,omitempty"`
+}
+
+type KeycloakAPIAuthenticationExecution struct {
+	// Authenticator
+	// +optional
+	Authenticator string `json:"authenticator,omitempty"`
+
+	// Authenticator Config
+	// +optional
+	AuthenticatorConfig string `json:"authenticatorConfig,omitempty"`
+
+	// Authenticator flow
+	// +optional
+	AuthenticatorFlow bool `json:"authenticatorFlow,omitempty"`
+
+	// Flow Alias
+	// +optional
+	FlowAlias string `json:"flowAlias,omitempty"`
+
+	// Priority
+	// +optional
+	Priority *int32 `json:"priority,omitempty"`
```

Comment:
Should we keep fields like this? Isn't priority based on the order in the file more natural, or should two executions have the same priority?

Reply:
I think it's fine. Let's leave it for now.